### ISSUE TITLE
[VULKAN] Segmentation fault when Context Shifting erase tokens

### ISSUE THREAD

**Issue by TheBill2001** [LostRuins/koboldcpp#647]
Love the Vulkan backend. It seem to be processing prompt faster than CuBLAS, which I love. But slower when generating. I ran some test here #646. However, a segmentation fault occurred when Context Shifting erased tokens, ie:

```bash
[Context Shifting: Erased 49 tokens at position 2719]
```

The stack trace recorded from systemd is very short:
```bash
           PID: 384139 (koboldcpp-linux)
           UID: 1000 (tuantran1632001)
           GID: 1000 (tuantran1632001)
        Signal: 11 (SEGV)
     Timestamp: Sun 2024-01-28 22:31:16 +07 (3min 3s ago)
  Command Line: ./koboldcpp-linux-x64 --model ./models/LLaMA2-13B-Psyfighter2.Q4_K_M.gguf --highpriority --contextsize 8192 --skiplauncher --gpulayers 7 --threads 12 --usevulkan
    Executable: /home/tuantran1632001/koboldcpp/koboldcpp-linux-x64
 Control Group: /user.slice/user-1000.slice/session-1.scope
          Unit: session-1.scope
         Slice: user-1000.slice
       Session: 1
     Owner UID: 1000 (tuantran1632001)
       Boot ID: f32038df516546659a2e6df638eaef9c
    Machine ID: 756efd162e8f4a8398f34f42f5a609c4
      Hostname: billarch-pc
       Storage: /var/lib/systemd/coredump/core.koboldcpp-linux.1000.f32038df516546659a2e6df638eaef9c.384139.1706455876000000.zst (present)
  Size on Disk: 53.3K
       Message: Process 384139 (koboldcpp-linux) of user 1000 dumped core.
                 
                Module koboldcpp-linux-x64 without build-id.
                Stack trace of thread 384139:
                #0  0x00007353c582283c n/a (libc.so.6 + 0x8e83c)
                #1  0x00007353c57d2668 raise (libc.so.6 + 0x3e668)
                #2  0x00005b5e062e3ec4 n/a (koboldcpp-linux-x64 + 0x3ec4)
                #3  0x00007353c57bbcd0 n/a (libc.so.6 + 0x27cd0)
                #4  0x00007353c57bbd8a __libc_start_main (libc.so.6 + 0x27d8a)
                #5  0x00005b5e062e20a1 n/a (koboldcpp-linux-x64 + 0x20a1)
                ELF object binary architecture: AMD x86-64

```

**Comment by PhialsBasement** [comment]
Ive been having the same issue, disabling the  context shifting seems to fix the issue and it doesnt appear to affect the performance or quality so id say do that till the guys figure it out

**Comment by LostRuins** [comment]
I can confirm this happens in llama.cpp vulkan too, it will segfault after attempting a context shift. This works fine if 0 GPU layers are offloaded.

**Comment by LostRuins** [comment]
Related : https://github.com/ggerganov/llama.cpp/issues/5179

**Comment by LostRuins** [comment]
Should be fixed now in v1.57!
